src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -17,7 +21,39 @@ public LineService(LineRepository lineRepository) {
     }
 
     public LineResponse saveLine(LineRequest request) {
-        Line persistLine = lineRepository.save(request.toLine());
-        return LineResponse.of(persistLine);
+        Line persistLine = new Line();
+        try {
+            persistLine = lineRepository.save(request.toLine());
+        } catch (Exception e) {
+            System.out.println(e.getMessage()); //log.error
+        } finally {
+            return LineResponse.of(persistLine);
+        }
+    }
+
+    public List<LineResponse> findByAll() {
+        List<Line> lines = lineRepository.findAll();
+        return lines.stream().map(line -> LineResponse.of(line)).collect(Collectors.toList());
+    }
+
+    public LineResponse findById(Long id) {
+        Line line = new Line();
+        try {
+            line = lineRepository.findById(id).get();
```

Comment:
제가 피드백을 명확히 전달하지 못한 것 같아 정정합니다.

`Optional.get()` 을 호출하면 항상 null이 반환되지 않을거라고 착각할 수도 있지 않을까요?
`Optional` 의 `get()` 내부에서 예외를 발생시키는 것 보다 아래와 같이 명시적으로 예외를 발생시키는게 어떨까요?

```suggestion
            line = lineRepository.findById(id).orElseThrow(...);
```

Reply:
아 제가 잘못 이해했었네요. 다음 step에서 적용하도록 하겠습니다!